common/buffers/src/main/java/io/helidon/common/buffers/FixedBufferData.java

```diff
@@ -129,7 +138,7 @@ public FixedBufferData write(int value) {
 
     @Override
     public int writeTo(ByteBuffer writeBuffer, int length) {
-        int toWrite = Math.min(writeBuffer.limit() - writeBuffer.position(), this.length - readPosition);
+        int toWrite = Math.min(writeBuffer.limit() - writeBuffer.position(), writePosition - readPosition);
```

Comment:
A bug we had before? Looks good now

Reply:
Yes, this was a bug. Luckily we never used `ByteBuffer` before